### ISSUE TITLE
Incorrect `vkBindBufferMemory2` best practice warning when using `vkGetDevice{Buffer/Image}MemoryRequirements`

### ISSUE THREAD

**Issue by phoekz** [KhronosGroup/Vulkan-ValidationLayers#5616]
**Environment:**
 - OS: Ubuntu 22
 - GPU: RTX 2060
 - SDK or header version if building from repo: 1.3.246 `ef7c246`
 - Options enabled (synchronization, best practices, etc.): best practices, debug printf, synchronization

**Describe the Issue**

Validation layers report `UNASSIGNED-BestPractices-vkBindBufferMemory-requirements-not-retrieved` and `UNASSIGNED-BestPractices-vkBindImageMemory-requirements-not-retrieved` if you bind device memory with memory requirements queried using `vkGetDeviceBufferMemoryRequirements` or `vkGetDeviceImageMemoryRequirements` . These functions are provided by Vulkan 1.3 as alternatives to `vkGetBufferMemoryRequirements` and `vkGetImageMemoryRequirements`.

Example warning:
```
type=VALIDATION
message_id_name=UNASSIGNED-BestPractices-vkBindBufferMemory-requirements-not-retrieved
message=Validation Warning: [ UNASSIGNED-BestPractices-vkBindBufferMemory-requirements-not-retrieved ] Object 0: handle = 0x55bb76411b80, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0xc82c4110 | vkBindBufferMemory2() pBindInfos[0]: Binding memory to VkBuffer 0x84c0580000000017[] but vkGetBufferMemoryRequirements() has not been called on that buffer.
```

**Expected behavior**

Best practices validation should not report this warning.

**Comment by juan-lunarg** [comment]
Thanks for reporting this. I should have a fix soon.

**Comment by juan-lunarg** [comment]
Duplicate of https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/4730

**Comment by juan-lunarg** [comment]
This will be fixed in the upcoming SDK release.
